src/network/networkd-link.c

```diff
@@ -242,6 +243,15 @@ void link_update_operstate(Link *link, bool also_update_master) {
         else
                 operstate = LINK_OPERSTATE_ENSLAVED;
 
+        /* Only determine online state for managed links with RequiredForOnline=yes */
+        if (!link->network || !link->network->required_for_online)
+                online_state = _LINK_ONLINE_STATE_INVALID;
+        else if (operstate >= link->network->required_operstate_for_online.min &&
+                 operstate <= link->network->required_operstate_for_online.max)
```

Comment:
Could you also take into account `ipv4/6_address_state` and `required_address_family`?